### DISCUSSION TITLE
A more pretty HTML output

### DISCUSSION THREAD

**Discussion by sahiljhawar** [rendercv/rendercv#96] [Ideas]
How feasible would it be to generate a `rst` file following which it can be built into a more pretty `HTML` webpage?

**Comment by sinaatalay** [comment]
I don't have any experience with `rst` files. Any guidance on this matter would be appreciated.

**Comment by sahiljhawar** [comment]
`rst` are like `md` but on steriods and sphinx loves `rst` files to generate web pages (the Python docs you see are mostly `rst` now a days)

**Comment by sahiljhawar** [comment]
Okay, so the script indeed generates `md`. This can be used to generate sphinx based html. I will try have a more detailed look into it.

**Comment by sahiljhawar** [comment]
@sinaatalay I tried to read the code and I was easily lost within the code. But I have a suggestion for pretty HTML, alongside creating the Grammarly HTML file, we can pass the generated markdown to sphinx and create a more better looking HTML doc (based on tempelate taken from the user). Of course this can be an optional dependency. Any thoughts?

**Comment by sinaatalay** [comment]
You don't need to read the code for this. When you use [`rendercv new`](https://docs.rendercv.com/user_guide/#getting-started-with-the-new-command), a folder called `markdown` will be generated. This folder contains the Markdown files, which are automatically converted into HTML. If you make changes to the contents of this folder, the corresponding HTML will be updated accordingly.

If you think RST will be a good choice, I suggest the following:

1.  Update the contents of the `markdown` folder as if they are `RST`.
2.  Then take the output (`John_Doe.md`) and rename it to `John_Doe.rst`. Then, use your favorite RST to HTML tool to see the outcome.

My thoughts about a pretty HTML:

HTML is just a markup language; the style and the design come from a CSS file. People can use CSS to make the existing HTML output a very nice-looking page and also modify the Markdown to modify the HTML output. I wouldn't want to ship a CSS with RenderCV because that would be another big thing to maintain, and LaTeX designs already need maintenance.

The HTML output's primary purpose is to enable users to easily copy and paste the CV's content into other software applications, such as Grammarly.

**Comment by sahiljhawar** [comment]
I will try the `rendercv new`. Maybe rst can be skipped since markdown is already there. But my rationale behind the HTML is to allows users to have a low-code web based CV which they don’t have to maintain separately. And with sphinx it cannot be any simpler. The formatting and everything is taken care of by the sphinx

**Comment by sinaatalay** [comment]
HTML output will take care of that. The CSS file doesn't need to be updated every time an HTML file changes; it's a one-time development.

If you could post an example of an RST CV and an example of HTML output under this issue, we can think of using RST as well.

**Comment by sahiljhawar** [comment]
User can of course setup the sphinx-doc manually if they want it (as I already did it). Most Python documentation (md and rst) are sphinx based

**Comment by sahiljhawar** [comment]
Implementing RST will be a humongous task, I think. But I will soon share the sphinx code (if I manage to integrate the Sphinx API with rendercv)

**Comment by sinaatalay** [comment]
It should be very easy; when you try out `rendercv new`, you will understand what I mean. It requires a maximum of 60 lines of work.

**Comment by sinaatalay** [comment]
I converted this into a discussion for now.

**Comment by sahiljhawar** [comment]
@sinaatalay I have an MWE for this

**Comment by sahiljhawar** [comment]
@sinaatalay 

clone this branch recursively :) : https://github.com/sahiljhawar/rendercv/tree/sphinx

and then run

```bash
pip install .
rendercv new test
rendercv render test_CV.yaml --generate-sphinx
```

after this this will create sub directories in the `rendercv_output` 

<img width="287" alt="image" src="https://github.com/sinaatalay/rendercv/assets/55475299/c3b82fe3-99b6-4165-b5f5-0349f73a0ac0">

`test_cv.html` is file of interest

**Comment by sahiljhawar** [comment]
if you plan to move forward with this, we can add option to specify the sphinx theme. I think that's the only option which is needed. And user can modify the `conf.py` as per their need. Once the `conf.py` is created it never gets overwritten.

**Comment by sahiljhawar** [comment]
Irrespective of that, your project is supercool, and I will soon move my CV from LaTeX to `rendercv` since it will be easier for me to publish this on the website

**Comment by sahiljhawar** [comment]
See here a live version of this CV : [sahiljhawar.me](https://sahiljhawar.me/)

**Comment by sahiljhawar** [comment]
With this, anyone can build CV using your program and ship it to their website. Atleast I am going to do this :)

**Comment by sinaatalay** [comment]
Thank you! I will have a look at my first availability.

**Comment by sahiljhawar** [comment]
Okay, I realised that people do not use sphinx for their personal website. They are more happy with Jekyll. Hence, adding sphinx will just be useless. Rest is up to you.

**Reply by sinaatalay** [reply]
Thank you. I will have a look this weekend and share my view on this.

**Reply by sinaatalay** [reply]
Thank you! I used your repository, and it works nicely.

We cannot default this to RenderCV because it adds

-  three big dependencies,
-  a new file,
-  more code

to the project and generates Sphinx documentation only for one purpose: **using the Sphinx's CSS file.**

Instead, we can do the following:

Add a pre-written CSS from a link at the beginning of the HTML file.

I have considered this approach before, but I was concerned that users might want to use plain HTML directly in their projects so that they can integrate it with their own CSS. However, I realized that Markdown can be easily converted to HTML in almost any environment; they can always use the original Markdown to get plain HTML. Therefore, in the next release, RenderCV will output aesthetically pleasing HTMLs with [github-markdown-css](https://github.com/sindresorhus/github-markdown-css).

Thank you for the idea!
